### PR TITLE
Allow for EAV varchar attributes to have leading zeros added/removed 

### PR DESCRIPTION
**Pull request by JeremyPlease**
### Description (*)

The previous Magento behavior was to not update EAV attributes that were numeric strings with leading zeros added or removed (ie. changing from `09` to `9`). This PR fixes this behaviour by removing the `is_numeric` check and only using strict comparison (`!==`) to determine if a value has changed.

### Fixed Issues (if relevant)

1. magento/magento2#7170: "Numeric-ish values not properly evaluated on varchar attributes"

### Manual testing scenarios (*)

1. Add a custom attribute named "num_string" with type of `varchar` to customers
2. Create a new customer and set "num_string" to "09"
3. Edit the newly created custer and set "num_string" to "9" and save

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green - [pending](https://travis-ci.org/magento/magento2/builds/451913066))

-----

Thanks to @serhiirepin for pointing out the solution in https://github.com/magento/magento2/issues/7170#issuecomment-404169043.

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @JeremyPlease. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.2-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by orlangur — CHANGES_REQUESTED** [review]
Nowadays all changes must be applied to `2.3-develop` first (see https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules). Please prepare a new pull request, this one should be `on hold` until the latter is merged.

Also, please cover changes with integration test.

**Comment by JeremyPlease** [comment]
Thanks for the quick review @orlangur!

New PR created: #19107. I'll close this PR so that a proper backport PR can be made later.

As for an integration test, could you please point me to some examples or documentation around testing an abstract class? It seems like most recently merged PRs don't have any new integration tests.

**Pull request closed without merging**
